src/renderer/helpers/api/local.js

```diff
@@ -482,9 +483,13 @@ export function parseLocalListPlaylist(playlist, author = undefined) {
       channelName = playlist.author.name
       channelId = playlist.author.id
     }
-  } else {
+  } else if (author) {
     channelName = author.name
     channelId = author.id
+  } else if (!isNullOrEmpty(playlist.author?.name)) {
```

Comment:
**question:** what is `isNullOrEmpty` doing differently from the default truthiness check? If it is doing anything, this should be before the `if (author)` check.

Reply:
There is a difference between `playlist.author` and the `author` parameter.

The logic in this function is:
1. Check if the playlist has a valid author object, if yes use it
2. Check if a fallback author was passed into the function, if yes use that
3. Check if the playlist has a partial/incomplete author object, if yes use it

The important bit is that we only want to use the incomplete playlist author object, if the caller didn't provide a fallback, we want to give the fallback higher priority than incomplete data (the fallback is used ont the channel page for example, where the playlists don't come with author information in them).

Reply:
Ah, I'm sorry, I totally missed the `playlist.` part! That makes sense. Same Q as before, though, on why we have a `isNullOrEmpty` function. Is it something having to do with type coercion?

Reply:
That function checks for nullish values and empty strings, it's useful because `0` counts as false as well, although I guess in this case that isn't a concern, so I can probably change it to just being `if (playlist.author?.name) {`, as empty strings are falsy but not nullish.

https://developer.mozilla.org/en-US/docs/Glossary/Nullish
https://developer.mozilla.org/en-US/docs/Glossary/Falsy

Reply:
Simplified the check now.

Reply:
Ah, ok, just was curious why we had it. That explanation would make sense, but looking at our existing uses, it seems to be "`!!` is yucky", which is fine lol